skrifa/src/outline/mod.rs

```diff
@@ -0,0 +1,448 @@
+//! Loading, scaling and hinting of glyph outlines.
+
+mod cff;
+mod error;
+mod glyf;
+mod native_hint;
+
+use read_fonts::{types::GlyphId, TableProvider};
+
+pub use error::DrawError;
+pub use native_hint::{NativeHinting, NativeHintingInstance};
+
+pub use read_fonts::types::Pen as OutlinePen;
+
+use super::{
+    instance::{LocationRef, NormalizedCoord, Size},
+    GLYF_COMPOSITE_RECURSION_LIMIT,
+};
+
+/// Source format for an outline glyph.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum OutlineGlyphFormat {
+    /// TrueType outlines sourced from the `glyf` table.
+    Glyf,
+    /// PostScript outlines sourced from the `CFF` table.
+    Cff,
+    /// PostScript outlines sourced from the `CFF2` table.
+    Cff2,
+}
+
+/// Specifies the hinting strategy.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+pub enum Hinting {
+    /// Hinting is disabled.
+    #[default]
+    None,
+    /// Will apply native embedded hints.
+    Native,
+}
+
+/// Information and adjusted metrics generated while drawing an outline glyph.
+#[derive(Copy, Clone, Default, Debug)]
+pub struct AdjustedMetrics {
+    /// True if the underlying glyph contains flags indicating the
+    /// presence of overlapping contours or components.
+    pub has_overlaps: bool,
+    /// If present, an adjusted left side bearing value generated by the
+    /// scaler.
+    pub lsb: Option<f32>,
+    /// If present, an adjusted advance width value generated by the
+    /// scaler.
+    pub advance_width: Option<f32>,
+}
+
+/// Options that define how a glyph is drawn to a pen.
+pub struct DrawSettings<'a> {
+    instance: DrawInstance<'a>,
+    memory: Option<&'a mut [u8]>,
+}
+
+impl<'a> DrawSettings<'a> {
+    /// Creates settings for an unhinted draw operation with the given size and
+    /// location in variation space.
+    pub fn new(size: Size, location: impl Into<LocationRef<'a>>) -> Self {
```

Comment:
Optional: I wonder if this should be more explicit, e.g. unhinted instead of new

Reply:
I lol’d. My first pass named this unhinted but I thought reviewers might not like that :)

Reply:
Renamed back to `unhinted` :)